config-model/src/main/java/com/yahoo/vespa/model/clients/ContainerDocumentApi.java

```diff
@@ -41,13 +41,17 @@ private static void addFeedHandler(ContainerCluster<?> cluster, Options options)
 
 
     private static void addRestApiHandler(ContainerCluster<?> cluster, Options options) {
-        var handler = newVespaClientHandler(
-                "com.yahoo.document.restapi.resource.RestApi", "/document/v1/*", options);
+        String handlerClassName = options.useNewRestapiHandler
+                ? "com.yahoo.document.restapi.resource.DocumentV1ApiHandler"
+                : "com.yahoo.document.restapi.resource.RestApi";
+        var handler = newVespaClientHandler(handlerClassName, "/document/v1/*", options);
         cluster.addComponent(handler);
-        var executor = new Threadpool(
-                "restapi-handler", cluster, options.restApiThreadpoolOptions, options.feedThreadPoolSizeFactor);
-        handler.inject(executor);
-        handler.addComponent(executor);
+        if (!options.useNewRestapiHandler) {
+            var executor = new Threadpool(
+                    "restapi-handler", cluster, options.restApiThreadpoolOptions, options.feedThreadPoolSizeFactor);
+            handler.inject(executor);
```

Comment:
Explicit injection 😱

Reply:
This is how the cool kids on the street do it